src/components/networks/utils.js

```diff
@@ -51,47 +46,37 @@ export function ipv4IsBroadcast(address, netMask) {
  * @param {string} address
  * @returns {boolean}
  */
-export function ipv4IsNetworkIdentifier(address, netMask) {
-    if (!validateNetmask(netMask))
+export function ipv4IsNetworkIdentifier(address, netmask) {
+    const prefix = validateNetmask(netmask);
+    if (prefix === null)
         return false;
-    const mask = netmaskConvert(netMask);
-    const subnet = ip.subnet(address, mask);
-
-    // user provided network identifier
-    return address === subnet.networkAddress;
+    return address === ipaddr.IPv4.networkAddressFromCIDR(`${address}/${prefix}`).toString();
 }
 
 /**
  * validates correctness of ipv4 prefix length or mask
  *
  * @param {string} prefixOrNetmask
- * @returns {boolean}
+ * @returns int prefix length, or null if invalid
  */
 export function validateNetmask(prefixOrNetmask) {
-    const netmaskParts = ["255", "254", "252", "248", "240", "224", "192", "128", "0"];
-    const parts = prefixOrNetmask.split('.');
-
-    // prefix length
-    if (parts.length === 1) {
-        if (!/^[0-9]+$/.test(parts[0].trim()))
-            return false;
-        const prefixLength = parseInt(parts[0], 10);
-        if (isNaN(prefixLength) || prefixLength < 1 || prefixLength > 31)
-            return false;
-
-        return true;
-    }
-
-    // netmask
-    if (!validateIpv4(prefixOrNetmask))
-        return false;
-
-    for (let i = 0; i < 4; i++) {
-        if (!(netmaskParts.includes(parts[i])))
-            return false;
+    if (/^[0-9]+$/.test(prefixOrNetmask)) {
+        // prefix
+        try {
+            const prefix = parseInt(prefixOrNetmask);
+            ipaddr.IPv4.subnetMaskFromPrefixLength(prefix);
+            return prefix;
+        } catch (_ex) {
+            return null;
+        }
+    } else {
+        // mask
+        try {
+            return ipaddr.IPv4.parse(prefixOrNetmask).prefixLengthFromSubnetMask();
+        } catch (_ex) {
+            return null;
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1455-20240222-112249-96a33f93-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)